### DISCUSSION TITLE
Clarification on various settings.

### DISCUSSION THREAD

**Discussion by devanteweary** [Feramance/qBitrr#96] [Q&A] [answered]
Hey, I had some questions I've been saving up while you've been working hard on the other issues.

I think I know most of these but wanted to get clarification on what these did. 
Thank you!

---

What's the difference between these? 
Is the search not part of the "reprocessing?"
```
[Settings]
LoopSleepTimer
SearchLoopDelay
```

Says it calls to update the queue. What does it mean by update the queue? 
What queue?
```
[Sonarr-TV]
RefreshDownloadsTimer = 5
```

How exactly does searching by year work? 
I noticed in the logs I've been sending, there's always something about searching a certain year and it says it adds a torrent (but it doesn't actually add anything). Just wondering how year search works in general and any cons to just disabling this.
```
[Sonarr-TV.EntrySearch]
SearchByYear = true
```

What does restarting the loop do? Just go on and search the next year? 
```
SearchAgainOnSearchCompletion = true
```

How does this one work?
```
UseTempForMissing = true
```

How slow is "slow?"
```
DoNotRemoveSlow
```

And finally, how does SearchOverseerrRequests work compared to qBitrr searching via Radarr (SearchMissing)?
Are they redundant?

**Comment by Feramance** [comment, marked as answer]
Hi,

To answer these qustions:


> What's the difference between these?
> Is the search not part of the "reprocessing?"

```
[Settings]
LoopSleepTimer - how often to loop through and check torrents
SearchLoopDelay - how often to send a search command to the arr apps
```


> Says it calls to update the queue. What does it mean by update the queue?
> What queue?

This refers to the queue in the Arr apps, how often to refresh so that the Arr apps recognise torrents.


> How exactly does searching by year work?
> I noticed in the logs I've been sending, there's always something about searching a certain year and it says it adds a torrent (but it doesn't > actually add anything). Just wondering how year search works in general and any cons to just disabling this.

This is a matter of preference. With the setting disabled qBitrr will grab your entire library from the Arr app, process the data it needs and starts a search for whatever is required. With it enabled it only processes the data it needs and searches for the apps of the spicified year. For example, say the first year is `1950` then it will grab all entries that were released in `1950`, gather the data required about them, then if needed search. Once completed it moves on to the next year that you have entries for. It doesn't search `1950,1951,1952...` qBitrr is smart and can tell you have entries only in years `1950,1976,1996...` for example.


> What does restarting the loop do? Just go on and search the next year?

This is simply to choose whether you only want it to perform the search commands once per run, this would mean that for qBitrr to send search commands again to the Arr apps, you would need to restart it (if the setting is disabled).


> How does this one work?
```
UseTempForMissing = true
```

This essentially allows you to set two quality profiles with different rules so that for example. In my case my main profile has a bunch of custom formats which lead to a ton of releases having a negative score below the minimum CF score set and this leaves some niche or unpopular movies to remain missing 'forever'. With this setting I could create a second quality profile with a minimum CF score at -9999999 which will allow negatively scored releases to be grabbed as I'd rather have something than nothing. qBitrr handling changing the quality profiles of movies listed as `Missing` and once it is no longer missing, it changes the quality profile back to the main one configured.


> How slow is "slow?"

Slow is defined on the config value below, so you define how slow is slow:
```
# Maximum allowed remaining ETA for torrent completion (in seconds: 3600 = 1 Hour)
# Maximum allowed remaining ETA for torrent completion (in seconds: 3600 = 1 Hour)
# Note that if you set the MaximumETA on a tracker basis that value is favoured over this value
MaximumETA = 604800
```


> And finally, how does SearchOverseerrRequests work compared to qBitrr searching via Radarr (SearchMissing)?
> Are they redundant?

Technically yes, they are redundant. It's an option for those who don't want to use the SearchMissing feature, but want to force search any Requests in Ombi/Overseerr
